src/lib/slp/dialogs/service_selection.rb

```diff
@@ -0,0 +1,163 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "ui/dialog"
+
+module Yast
+  module Dialogs
+    # This class implements a services selection dialog.
+    class ServiceSelection < ::UI::Dialog
+      Yast.import "UI"
+      Yast.import "Label"
+      Yast.import "Report"
+
+      # @return [Array<SlpServiceClass::Service] list of services to show
+      attr_reader :services
+      # @return [String] dialog's heading
+      attr_reader :heading
+      # @return [String] dialog's description
+      attr_reader :description
+      # @return [String] message to be shown when no service was selected
+      attr_reader :no_selected_msg
+      # @return [SlpServiceClass::Service] initially selected service
+      attr_reader :initial
+
+      # Run dialog
+      #
+      # The return value will be:
+      # * A service in case one was selected
+      # * :cancel symbol if the dialog was canceled
+      #
+      # @example Select some service
+      #   Yast::Dialogs::SelectionServiceDialog.run(services) #=> :scc
+      #     #=> #<Yast::SlpServiceClass::Service...>
+      #
+      # @example Press the 'cancel' button
+      #   Yast::Dialogs::SelectionServiceDialog.run(services) #=> :cancel
+      #
+      # @param services        [Array<SlpServiceClass::Service] list of services to show
+      # @param heading         [String] Dialog's heading
+      # @param description     [String] Dialog's description (to be shown on top of the list)
+      # @param no_selected_msg [String] Message to be shown when no service was selected
```

Comment:
Without knowing the implementation this does not make sense. I would expect the dialog simply does not let me proceed without selecting a service (unless I retreat with Cancel).